src/client/src/components/abstract/pieChart/index.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from "react";
+import { makePie, colors } from "./logic";
+import "./index.css";
+export default class index extends Component {
+  componentDidMount = () => {
+    const { sections, id } = this.props;
+    makePie(sections, id);
+  };
+  render() {
+    const { sections, id } = this.props;
+    return (
+      <div className="pie">
+        <svg className={`bar-chart-${id} bar-chart`} />
```

Comment:
@knanahassouna1 
You can use React's [ref](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) prop here as a solution for displaying multiple pie charts on-screen instead of your `id`.

`ref` lets you get and store a reference to an actual DOM node that your component instance has rendered.  Once you have that, you can use that directly in makePie, instead of having makePie search the dom for bar-chart-${id}.

So here, first you would grab a ref this component's svg element, like this:
```
<svg className='bar-chart' ref={(element) => this.svgElement = element} />
```
The syntax is a little weird, but you are basically giving React a callback that will be called once the component renders for the first time.  Inside that callback, `this` means this component.

Next, in componentDidMount() you could start using that element - you can pass it into makePie:
```
componentDidMount = () => {
    const { sections } = this.props;
    makePie(sections, this.svgElement);
};
```

Last, you can use that element inside makePie without much changing - you can pass the node directly into d3.select() to make d3 use that particular dom node:
```
export const makePie = (sections, svgElement) => {
  ...
  const svg = d3
    .select(svgElement)
  ...
```

That should work, and then you don't need the `id` prop any more (so then you don't have to worry about keeping ids unique).  Let me know what you think.